core/survey/_survey/surveyDependencies.js

```diff
@@ -16,10 +16,11 @@ const keys = {
 
 const _getDeps = (type, nodeDefUuid) => R.pathOr([], [type, nodeDefUuid])
 
-const _addDep = (type, nodeDefUuid, nodeDefDepUuid) => (graph) =>
-  R.pipe(_getDeps(type, nodeDefUuid), R.append(nodeDefDepUuid), (dep) =>
-    ObjectUtils.setInPath([type, nodeDefUuid], dep)(graph)
-  )(graph)
+const _addDep = (type, nodeDefUuid, nodeDefDepUuid) => (graph) => {
+  const deps = _getDeps(type, nodeDefUuid)(graph)
+  const depsUpdated = [...deps, nodeDefDepUuid]
```

Comment:
much easier to read:

take a look of:
https://stackoverflow.com/questions/48865710/spread-operator-vs-array-concat